src/log.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from pathlib import Path
+
+
+def setup_logger(log_file: Path = None, level=logging.INFO) -> logging.Logger:
+    """
+    Sets up the logging configuration for the application.
+
+    Args:
+        log_file (Path): Optional path to a log file where logs will be saved.
+        level (int): Logging level (INFO, DEBUG, etc.).
+
+    Returns:
+        logging.Logger: Configured logger instance.
+    """
+    if log_file is not None:
+        log_file.parent.mkdir(exist_ok=True)
+
+    logger = logging.getLogger(__name__)
+    logger.setLevel(level)
+
+    # Create console handler
+    console_handler = logging.StreamHandler()
```

Comment:
if you're down to take on a dependency, def recommend the [rich logger](https://github.com/Aharoni-Lab/miniscope-io/blob/d4e45971db290bee72cbfbaf7a86cd214b8279de/miniscope_io/logging.py#L94-L100)

Reply:
Also check if this double-logs, you might want to set `propagate = False` if `logger.parent == logging.RootLogger`